src/pythie_serving/lightgbm_wrapper.py

```diff
@@ -1,61 +1,92 @@
+        self,
+        *,
+        logger: logging.Logger,
+        model_server_config: model_server_config_pb2.ModelServerConfig,
+    ):
         self.logger = logger
         self.model_map = {}
         for model_config in model_server_config.model_config_list.config:
-            with open(model_config.base_path, 'rb') as opened_model:
+            with open(model_config.base_path, "rb") as opened_model:
                 model = pickle.load(opened_model)
 
                 if isinstance(model, Booster):
                     feature_names = model.feature_name()
                     best_iteration = model.best_iteration
                 else:
                     feature_names = model.feature_names
-                    best_iteration = getattr(model, 'best_iteration', None)
+                    best_iteration = getattr(model, "best_iteration", None)
 
-                self.model_map[model_config.name] = {'model': model, 'feature_names': feature_names,
-                                                     'best_iteration': best_iteration}
+                self.model_map[model_config.name] = {
+                    "model": model,
+                    "feature_names": feature_names,
+                    "best_iteration": best_iteration,
+                }
 
     def Predict(self, request: predict_pb2.PredictRequest, context: grpc.RpcContext):
         model_name = request.model_spec.name
         if model_name not in self.model_map:
-            raise PythieServingException(f'Unknown model: {model_name}. This pythie-serving instance can only '
-                                         f'serve one of the following: {",".join(self.model_map.keys())}')
+            raise PythieServingException(
+                f"Unknown model: {model_name}. This pythie-serving instance can only "
+                f'serve one of the following: {",".join(self.model_map.keys())}'
+            )
 
         model_dict = self.model_map[model_name]
-
-        features_names = model_dict['feature_names']
+        model = model_dict["model"]
+        features_names = model_dict["feature_names"]
+        pd_categorical_features = []
         samples = None
         for feature_name in features_names:
             if feature_name not in request.inputs:
-                raise PythieServingException(f'{feature_name} not set in the predict request')
+                raise PythieServingException(
+                    f"{feature_name} not set in the predict request"
+                )
 
             nd_array = make_ndarray_from_tensor(request.inputs[feature_name])
+            # get features categories from model if feature dtype is object ("string")
+            is_pd_categorical = False
+            feature_categories = []
+            if nd_array.dtype == object:
+                is_pd_categorical = True
+                # lgbm save categories in the same order categorical features appear in model.feature_name()
+                feature_categories = model.pandas_categorical[
```

Comment:
instead of computing pd_categorical_features at every inference, should you not compute this only once at init() time and save it in  self.model_map